lib/medtronic/processData.js

```diff
@@ -426,6 +464,84 @@ function buildBGRecords(records) {
   return postrecords;
 }
 
+function buildCGMRecords(events) {
+
+  var postrecords = [];
+  var start = null;
+  var recordsSinceTimestamp = null;
+
+  // we have to work through the CGM records in reverse, as sensor timestamps
+  // affect previous values and are subject to time changes
+
+  for( var i = events.length - 1 ; i >= 0; i-- ) {
+    var event = events[i];
+
+    if(event.jsDate != false) {
+      console.log('CGM event:',sundial.formatDeviceTime(event.jsDate), event.type.name, common.bytes2hex([event.head]),common.bytes2hex(event.date),common.bytes2hex(event.body),common.bytes2hex(event.descriptor) );
+
+      if(event.type.value === CBG_RECORD_TYPES.SENSOR_TIMESTAMP.value) {
+        start = event.jsDate;
+        recordsSinceTimestamp = 0;
+      }
+
+      if(event.body) {
+        var offset = event.body.length;
+        var index = event.index - offset;
+
+        while(offset >= 0) {
+
+          var isig = struct.extractShort(event.descriptor, offset * 2);
```

Comment:
Agreed, I'm renaming `isig` to `descriptor`. The structure doesn't really lend itself to being defined in a packet diagram, so I'm adding a description of the structure as a comment.